tests/Unit/AccountTest.php

```diff
@@ -540,18 +548,20 @@ public function test_it_skips_default_relationship_types_table_for_types_already
 
     public function test_it_retrieves_yearly_call_statistics()
     {
-        $account = factory('App\Account')->create([]);
-        $contact = factory('App\Call', 4)->create([
-            'account_id' => $account->id,
+        $contact = factory('App\Contact')->create();
+        $calls = factory('App\Call', 4)->create([
+            'account_id' => $contact->account_id,
+            'contact_id' => $contact->id,
             'called_at' => '2018-03-02',
         ]);
 
-        $contact = factory('App\Call', 2)->create([
-            'account_id' => $account->id,
+        $calls = factory('App\Call', 2)->create([
+            'account_id' => $contact->account_id,
+            'contact_id' => $contact->id,
             'called_at' => '1992-03-02',
         ]);
 
-        $statistics = $account->getYearlyCallStatistics();
+        $statistics = $contact->account->getYearlyCallStatistics();
```

Comment:
So here you chose to keep `$contact->account` 😀 Why?

Reply:
I'm not sure I understand what you mean. In this case we generate a contact instead of an account, which has an account generated at the same time. This is necessary since we need to assign both the contact_id and account_id, which wasn't done previously.

Reply:
I mean that above in this file, you've assigned `$account = $contact->account;` and then use `$account` in the methods. 

In this case you've kelp `$contact->account`

Reply:
Oh right. Good observation. I think I just tried to modify as little code as possible. As you can see from the class, there's a lot that could be changed to make the style more uniform (use classes everywhere instead of strings, use `$account` instead of `$contact->account`, etc.).